src/network/message.rs

```diff
@@ -66,12 +74,10 @@ impl Encodable for CommandString {
     fn consensus_encode<S: io::Write>(
         &self,
         s: S,
-    ) -> Result<usize, encode::Error> {
+    ) -> Result<usize, io::Error> {
         let mut rawbytes = [0u8; 12];
         let strbytes = self.0.as_bytes();
-        if strbytes.len() > 12 {
-            return Err(encode::Error::UnrecognizedNetworkCommand(self.0.clone().into_owned()));
-        }
+        debug_assert!(strbytes.len() <= 12);
         rawbytes[..strbytes.len()].clone_from_slice(&strbytes[..]);
```

Comment:
FWIW the second line will always do bounds check + panic if `strbytes` is more than 12 bytes

Reply:
Removed the debug_assert then.